### DISCUSSION TITLE
Custom Fonts not working

### DISCUSSION THREAD

**Discussion by MehrshadHeisenberg** [tailwindlabs/tailwindcss#13583] [Help] [answered]
Hi everyone
I am working on a project with React, Vite and Tailwindcss

When I try to add a custom font with font-face it seems that tailwindcss can not find it and show it properly.

I have tried anything I could think of and as of now I have my index.css file in the src folder which in there I have defined my fonts in the base layer using @font-face

Also in my tailwind.config,js file I have extended the font family that I need.

But when I use it in App.jsx it does not work and does not show the font that I have defined using font-face

I would really appreciate your help

Also here is the repo so you can check it out yourself:
[The related repo](https://github.com/MehrshadHeisenberg3/React-MultiStep-Form)

**Comment by wongjn** [comment, marked as answer]
As per the syntax for `src` in `@font-face` according [to MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/src#formal_syntax), you'd want `truetype`, not `ttf` for the `format()` values:

```diff
 @layer base {
   @font-face {
     font-family: "Ubuntu";
     font-weight: 400;
-    src: url("../public/fonts/Ubuntu-Regular.ttf") format("ttf");
+    src: url("../public/fonts/Ubuntu-Regular.ttf") format("truetype");
   }
 
   @font-face {
     font-family: "Ubuntu";
     font-weight: 500;
-    src: url("../public/fonts/Ubuntu-Medium.ttf") format("ttf");
+    src: url("../public/fonts/Ubuntu-Medium.ttf") format("truetype");
   }
 
   @font-face {
     font-family: "Ubuntu";
     font-weight: 700;
-    src: url("../public/fonts/Ubuntu-Bold.ttf") format("ttf");
+    src: url("../public/fonts/Ubuntu-Bold.ttf") format("truetype");
   }
```

You may also wish to consider changing the paths to the recommended `/` instead of `../public`:

```diff
 @layer base {
   @font-face {
     font-family: "Ubuntu";
     font-weight: 400;
-    src: url("../public/fonts/Ubuntu-Regular.ttf") format("truetype");
+    src: url("/fonts/Ubuntu-Regular.ttf") format("truetype");
   }
 
   @font-face {
     font-family: "Ubuntu";
     font-weight: 500;
-    src: url("../public/fonts/Ubuntu-Medium.ttf") format("truetype");
+    src: url("/fonts/Ubuntu-Medium.ttf") format("truetype");
   }
 
   @font-face {
     font-family: "Ubuntu";
     font-weight: 700;
-    src: url("../public/fonts/Ubuntu-Bold.ttf") format("truetype");
+    src: url("/fonts/Ubuntu-Bold.ttf") format("truetype");
   }
```
Or move them inside the `src` folder and reference them there.

**Reply by MehrshadHeisenberg** [reply]
Thanks mate, I'll try it and let you know

But why should I reference the fonts like that?

**Reply by wongjn** [reply]
As per [the documentation](https://vitejs.dev/guide/assets.html#the-public-directory):

> - You should always reference `public` assets using root absolute path - for example, `public/icon.png` should be referenced in source code as `/icon.png`.

**Reply by agcodex01** [reply]
Thanks @wongjn , it works for me
